microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/CircuitBreakerHelper.java

```diff
@@ -252,27 +282,24 @@ long getInStateNanos(State queryState) {
     }
 
     /**
-     * Ensure that our internal state matches Hystrix when a breaker in OPEN
-     * state. For some reason Hystrix does not set the breaker in OPEN state
-     * right away, and calling {@link HystrixCommand#isCircuitBreakerOpen()}
-     * appears to fix the problem.
+     * Force Hystrix's circuit breaker into an open state.
      */
-    void ensureConsistentState() {
-        if (getState() == State.OPEN_MP) {
-            long delayTotal = 0L;
-            while (!command.isCircuitBreakerOpen() && delayTotal < MAX_DELAY_CIRCUIT_BREAKER_OPEN) {
-                long delayPeriod = MAX_DELAY_CIRCUIT_BREAKER_OPEN / 10;
-                try {
-                    LOGGER.fine("Waiting for Hystrix to open circuit breaker (" + delayPeriod + " ms)");
-                    Thread.sleep(delayPeriod);
-                } catch (InterruptedException e) {
-                    // falls through
-                }
-                delayTotal += delayPeriod;
-            }
-            if (delayTotal >= MAX_DELAY_CIRCUIT_BREAKER_OPEN) {
-                throw new InternalError("Inconsistent state for circuit breaker in " + command.getCommandKey());
-            }
+    private void openBreaker() {
+        if (!command.isCircuitBreakerOpen()) {
+            ConfigurationManager.getConfigInstance().setProperty(
+                    String.format(FORCE_OPEN, command.getCommandKey()), "true");
+        }
```

Comment:
The revised `closeBreaker` method below sets `FORCE_OPEN` to false as well as `FORCE_CLOSED` to true.

Should this `openBreaker` method set `FORCE_CLOSED` to false as well as `FORCE_OPEN` to true?

I'm not very familiar with Hystrix so maybe I'm missing something.

Reply:
It's not necessary because if both are defined, FORCE_OPEN takes precedence according to their documentation. See:

https://github.com/Netflix/Hystrix/wiki/Configuration#circuitBreaker.forceOpen